R/estimateAlpha.R

```diff
@@ -0,0 +1,247 @@
+#'   calculation of the sample-wise estimates.
+#'   
+#' @param index a \code{character} vector, specifying the alpha diversity measures
+#'   to be calculated
+#'   
+#' @param name a name for the column(s) of the colData the results should be
+#'   stored in. By default this will use the original names of the calculated
+#'   indices specifying the alpha diversity measures used.
+#'   
+#' @param ... optional arguments.
+#' 
+#' 
+#' @param rarify logical scalar: Should the alpha diversity measures be estimated
+#'   using rarefaction? (default: \code{FALSE})
+#' 
+#' @param seed a single \code{integer} value as the seed used for the nround
+#'  rarefaction.
+#' 
+#' @param n.iter a single \code{integer} value for the number of rarefaction
+#' rounds.
+#' 
+#' @param rarefaction_depth a \code{double} value as for the minimim size or 
+#' rarefaction_depth. (default: \code{min(colSums(assay(x, "counts")), na.rm = TRUE)})
+#' 
+#' @return \code{x} with additional \code{\link{colData}} named after the index 
+#' used.
+#' 
+#' @examples
+#' 
+#' data("GlobalPatterns")
+#' tse <- GlobalPatterns
+#' 
+#' # Calculate the default Shannon index with no rarefaction
+#' tse <- estimateAlpha(tse, assay.type = "counts", index = "shannon")
+#' 
+#' # Shows the estimated Shannon index
+#' colData(tse)$shannon_diversity
+#'
+#'# Calculate observed richness with 10 rarefaction rounds
+#' tse <- estimateAlpha(tse, assay.type = "counts", index = "observed_richness",
+#' rarify=TRUE, n.iter=10)
+#' 
+#' # Shows the estimated observed richness
+#' colData(tse)$observed_richness
+#' 
+#' @importFrom dplyr %>% 
+#' 
+#' @rdname estimateAlpha
+#' @export
+estimateAlpha <- function(x, assay.type = "counts",
+                          index = c("coverage_diversity", "fisher_diversity",
+                                    "faith_diversity", "gini_simpson_diversity",
+                                    "inverse_simpson_diversity",
+                                    "log_modulo_skewness_diversity", "shannon_diversity",
+                                    "absolute_dominance", "dbp_dominance",
+                                    "core_abundance_dominance", "gini_dominance",
+                                    "dmn_dominance", "relative_dominance",
+                                    "simpson_lambda_dominance",
+                                    "camargo_evenness", "pielou_evenness",
+                                    "simpson_evenness", "evar_evenness",
+                                    "bulla_evenness",
+                                    "ace_richness", "chao1_richness", "hill_richness",
+                                    "observed_richness"),
+                          name = index,
+                          ...,
+                          rarify=FALSE,
+                          seed = 123,
+                          n.iter=10,
+                          rarefaction_depth=min(colSums(assay(x, assay.type)), na.rm = TRUE)){
+    # checks
```

Comment:
more explanations needed; add a comment line for each substantial step below to help reader understand what it does. Check other functions for examples.